pkg/kubectl/cmd/set/helper.go

```diff
@@ -112,40 +112,49 @@ type Patch struct {
 	Patch  []byte
 }
 
+// CalculatePatch calls the mutation function on the provided info object, and generates a strategic merge patch for
+// the changes in the object. Encoder must be able to encode the info into the appropriate destination type.
+// This function returns whatever mutateFn returns.
+func CalculatePatch(patch *Patch, encoder runtime.Encoder, mutateFn func(*resource.Info) (bool, error)) bool {
```

Comment:
Can this take an `Info` and return `[]byte, error` rather than mutating input?  that gives us a more "normal" API?